[Pages] mailchannels review (#12682)
cloudflare/cloudflare-docs: content/pages/functions/plugins/mailchannels.md

For more information about MailChannels and the options they support, refer to [the documentation](https://mailchannels.zendesk.com/hc/en-us/articles/4565898358413-Sending-Email-from-Cloudflare-Workers-using-MailChannels-Send-API).

## Enable MailChannels for your account - Domain Lockdown

To enable MailChannels to send emails on your behalf through Cloudflare Pages Functions, you have to setup a [Domain Lockdown DNS record](https://support.mailchannels.com/hc/en-us/articles/16918954360845-Secure-your-domain-name-against-spoofing-with-Domain-Lockdown-) for the domain you are sending emails from.

For most cases, setting up a Domain Lockdown DNS record involves adding a `TXT` entry to your DNS. Refer to the following steps to set up a Domain Lockdown DNS record.

> First, find your account's `workers.dev` subdomain:

Reviewer: I believe there is an error in this portion of the docs. This did not work for me. I kept getting email send error messages. 

What did work however was using the value from the pages application:
`Whatever.pages.dev`
and my custom domain `mydomain.com`

I am not the only one, see:
https://stackoverflow.com/a/77041550/275016

A lot of people on the internet are confused and having problems. Please re-evalute this part of the docs 🙏

3. Create a DNS record with the content of the generated `dkim_record.txt` file content.

Next, look in your generated `dkim_record.txt` file for your DKIM credentials, and add them to your website in the Cloudflare dashboard. Follow the steps below:

1. In Account Home, select the website you would like to add a DKIM record.
2. In the menu on the left select **DNS** > **Records** > **Add Record**.
3. Add the following TXT DNS record:

{{<example>}}

| Type | Name                      | Content                                  |
|------|---------------------------|------------------------------------------|
| TXT  | `mailchannels._domainkey` | Insert the contents of `dkim_record.txt` | 

{{</example>}}

{{<Aside type= "note" header="Use a different selector">}}

> You can substitute `mailchannels.domainkey` for any name with the format `<selector_key>._domainkey`. You can choose any value as the selector, as long as it is permitted as a DNS hostname (that is, all lowercase letters, numbers and hyphens).

Reviewer 2: ```suggestion
You can substitute `mailchannels._domainkey` for any name with the format `<selector_key>._domainkey`. You can choose any value as the selector, as long as it is permitted as a DNS hostname (that is, all lowercase letters, numbers, and hyphens).
```

```typescript
---
filename: functions/_middleware.ts
highlight: [7-12]
---
import mailChannelsPlugin from "@cloudflare/pages-plugin-mailchannels";

export const onRequest: PagesFunction = (context) => mailChannelsPlugin({
  personalizations: [
    {
      to: [{ name: "Some User", email: "someuser@example.com" }],

>       // This value has to be the domain you added DKIM records to and where you're sending your email from

Reviewer 2: ```suggestion
      // This value has to be the domain you added DKIM records to and where you are sending your email from
```

      "dkim_domain": "mydomain.com", 
      // This value has be the same as the selector you chose for your DKIM record name

>       // For example, use "mailchannels" if you used mailchannels._domainkey as your record name

Reviewer 2: ```suggestion
      // For example, use "mailchannels" if you used "mailchannels._domainkey" as your record name
```

      "dkim_selector": "mailchannels", 
      "dkim_private_key": context.env.DKIM_PRIVATE_KEY
    },
  ],
  from: {
    name: "ACME Support",

>     // The domain of your `from` address must be the same as the domain you setup MailChannels Domain Lockdown for

Reviewer 2: ```suggestion
    // The domain of your `from` address must be the same as the domain you set up MailChannels Domain Lockdown for
```
